technical-guide/developer/ui.md

```diff
@@ -660,3 +660,84 @@ You can use Storybook's `<Canvas>` element to showcase specific stories to enric
 When including codeblocks, please add code in Clojure syntax (not JSX).
 
 You can find an example MDX file in the [Buttons docs](https://hourly.penpot.dev/storybook/?path=/docs/buttons-docs--docs).
+
+### Replacing and old component
+
+**Create screenshots for visual tests BEFORE making any changes.**
+
+Run the tests in the DS project. On TMUX, stop Shadow and run:
+```bash
+clojure -M:dev:shadow-cljs release main
+```
+This will package the frontend in release mode so the tests run faster.
+
+In your terminal, in the frontend folder, run:
+```bash
+  npx playwright test --ui 
+```
+This will open the test runner UI. Select the DS project and run the tests.
+
+![Playwright UI](/img/tech-guide/playwright-projects.webp)
+The first time they'll fail because there are no screenshots yet, but the second time, they should pass.
+
+**Import the new component**
+
+In the selected file add the new namespace from the `ds` folder in alphabetical order:
+
+```clojure
+[app.main.ui.ds.tab-switcher :refer [tab-switcher*]]
+...
+
+[:> tab-switcher* {}]
+```
+
+> **⚠️ NOTE**:
+Pay attention to the use of `:>`.
+We need to pass props dynamically to create a native DOM element. Note that props are always JS plain objects, and we follow React naming conventions. We’re building a reusable component, probably wrapping one or more native virtual DOM elements, and we just want to control a subset of the props, while passing the rest unchanged to the next native element.
+
+<small>Extract from the [Rumext documentation](https://funcool.github.io/rumext/latest/user-guide.html)<small>
+
+Additionally, Rumext provides a way to ensure destructuring adheres to React's prop key conventions. You can use metadata `{::mf/props :react}` or add a `*` suffix to the component name. This allows you to use Lisp-case for keys, and the macro will auto-generate camelCase for props, aligning with React's conventions.
+
+At this point, the component might not load because of a props validation error.
```

Comment:
I feel this is a bit redundant with the existing documentation. 

I feel we can just say that something in the lines of "components with a `*` suffix are meant to be used with the `[:>` handler. Please refer to Rumext User Guide (link) for more information".